src/intrusive.rs

```diff
@@ -670,117 +734,57 @@ impl<R: ScopedRawMutex> StorageList<R> {
-fn set_counter(
-    ls: &mut MutexGuard<'_, List<NodeHeader>, impl ScopedRawMutex>,
-    new_counter: Counter,
-) {
-    debug!("Setting counter for list to: {}", new_counter);
-    // Update counters before writing the list to flash
-    // We do this before serialization because that may fail and
-    // we want to avoid diverging counter values...
-    for mut hdrptr in ls.iter_raw() {
-        let header = unsafe { hdrptr.as_mut() };
-        header.counter.replace(new_counter);
-    }
-}
-
-/// Writes a write confirmation block to flash memory to mark a successful write operation for
-/// the specified counter.
-///
-/// This function creates and writes a special "write_confirm" block to flash that serves as
-/// a marker indicating that a write operation with the given counter value has been successful.
-/// The write_confirm block consists of an all-zeros key followed by the counter value.
-///
-/// This confirmation mechanism helps distinguish between complete and incomplete write operations
-/// during recovery, allowing the system to determine which data entries are valid and which may
-/// have been interrupted mid-write.
-///
-/// # Arguments
-/// * `flash` - A mutable reference to the Flash storage device and address range to write to
-/// * `counter` - The counter value associated with the write operation being confirmed.
-///
-/// # Returns
-/// * `Ok(())` - If the write confirmation block was successfully written to flash
-/// * `Err(LoadStoreError::FlashWrite)` - If the flash write operation failed
-async fn confirm_write<F: MultiwriteNorFlash>(
-    flash: &mut Flash<F, impl CacheImpl>,
-    counter: Counter,
-) -> Result<(), LoadStoreError<F::Error>> {
-    // Assemble a write_confirm block consisting of an all-zeros key and the counter
-    let write_confirm = &mut [0u8; KEY_LEN + 1];
-    write_confirm[KEY_LEN] = counter.0;
-    // Try writing to flash
-    flash
-        .push(write_confirm)
-        .await
-        .map_err(LoadStoreError::FlashWrite)
-}
+// /// Removes old flash entries until reaching the newly written list with the specified counter.
+// ///
+// /// This function iterates through flash storage, removing outdated entries by popping them
+// /// from the queue until it encounters an entry with the target counter value, which indicates
+// /// the beginning of the newly written list.
+// ///
+// /// # Arguments
+// /// * `flash` - The flash storage device and address range to clean up
+// /// * `counter` - The counter value that marks the start of the newly written list
+// /// * `serde_buf` - Buffer used for flash operations (must be large enough for any flash item)
+// ///
+// /// # Returns
+// /// * `Ok(())` - If cleanup completed successfully
+// /// * `Err(LoadStoreError::FlashRead)` - If reading from flash fails
+// /// * `Err(LoadStoreError::FlashWrite)` - If removing old items fails
+// async fn delete_old_items<F: MultiwriteNorFlash>(
```

Comment:
noting that this is dead code, but I haven't removed it yet because I haven't found a home for the "do garbage collection of old elems/write reports" logic yet.